pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/PartitionSegmentPruner.java

```diff
@@ -18,266 +18,5 @@
-    FilterKind filterKind = FilterKind.valueOf(function.getOperator());
-    List<Expression> operands = function.getOperands();
-    switch (filterKind) {
-      case AND:
-        for (Expression child : operands) {
-          if (!isPartitionMatch(child, partitionInfo)) {
-            return false;
-          }
-        }
-        return true;
-      case OR:
-        for (Expression child : operands) {
-          if (isPartitionMatch(child, partitionInfo)) {
-            return true;
-          }
-        }
-        return false;
-      case EQUALS: {
-        Identifier identifier = operands.get(0).getIdentifier();
-        if (identifier != null && identifier.getName().equals(_partitionColumn)) {
-          return partitionInfo._partitions.contains(
-              partitionInfo._partitionFunction.getPartition(operands.get(1).getLiteral().getFieldValue().toString()));
-        } else {
-          return true;
-        }
-      }
-      case IN: {
-        Identifier identifier = operands.get(0).getIdentifier();
-        if (identifier != null && identifier.getName().equals(_partitionColumn)) {
-          int numOperands = operands.size();
-          for (int i = 1; i < numOperands; i++) {
-            if (partitionInfo._partitions.contains(partitionInfo._partitionFunction.getPartition(
-                operands.get(i).getLiteral().getFieldValue().toString()))) {
-              return true;
-            }
-          }
-          return false;
-        } else {
-          return true;
-        }
-      }
-      default:
-        return true;
-    }
-  }
-
-  @Deprecated
-  private boolean isPartitionMatch(FilterQueryTree filterQueryTree, PartitionInfo partitionInfo) {
-    switch (filterQueryTree.getOperator()) {
-      case AND:
-        for (FilterQueryTree child : filterQueryTree.getChildren()) {
-          if (!isPartitionMatch(child, partitionInfo)) {
-            return false;
-          }
-        }
-        return true;
-      case OR:
-        for (FilterQueryTree child : filterQueryTree.getChildren()) {
-          if (isPartitionMatch(child, partitionInfo)) {
-            return true;
-          }
-        }
-        return false;
-      case EQUALITY:
-      case IN:
-        if (filterQueryTree.getColumn().equals(_partitionColumn)) {
-          for (String value : filterQueryTree.getValue()) {
-            if (partitionInfo._partitions.contains(partitionInfo._partitionFunction.getPartition(value))) {
-              return true;
-            }
-          }
-          return false;
-        }
-        return true;
-      default:
-        return true;
-    }
-  }
-
-  private static class PartitionInfo {
-    final PartitionFunction _partitionFunction;
-    final Set<Integer> _partitions;
-
-    PartitionInfo(PartitionFunction partitionFunction, Set<Integer> partitions) {
-      _partitionFunction = partitionFunction;
-      _partitions = partitions;
-    }
-  }
+public interface PartitionSegmentPruner extends SegmentPruner {
```

Comment:
Let's remove this file since this extra interface does not provide extra value (it should show as a rename from `PartitionSegmentPruner` to `SinglePartitionColumnSegmentPruner`)